.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -1105,6 +1108,205 @@ stages:
       condition: succeededOrFailed()
       continueOnError: true
 
+- stage: asan_tests_windows_profiler
+  condition: > 
+    and(
+      succeeded(), 
+      eq(variables['isBenchmarksOnlyBuild'], 'False'),
+      eq(dependencies.generate_variables.outputs['generate_variables_job.generate_variables_step.IsProfilerChanged'], 'True')
+    )
+  dependsOn: [master_commit_id, generate_variables, build_windows_tracer, build_windows_profiler]
+  variables:
+    masterCommitId: $[ stageDependencies.master_commit_id.fetch.outputs['set_sha.master']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [Win]
+
+  - job: Win
+    pool:
+      vmImage: windows-2019
+    timeoutInMinutes: 60 #default value
+    strategy:
+      matrix:
+        x86:
+          platform: x86
+        x64:
+          platform: x64
+    variables:
+      USE_NATIVE_LOADER: true
+    
+    # Enable the Datadog Agent service for this job
+    services:
+      dd_agent: dd_agent
+
+    steps:
+    - template: steps/install-dotnet-sdks.yml
+
+    - task: DownloadPipelineArtifact@2
+      displayName: Download windows profiler home
+      inputs:
+        artifact: windows-profiler-binaries
+        path: $(System.DefaultWorkingDirectory)/profiler/_build
+
+    - template: steps/restore-working-directory.yml
+      parameters:
+        artifact: build-windows-working-directory
+
+    - task: MSBuild@1
+      inputs:
+        solution: profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/Datadog.Profiler.Native.Windows.WithTests.proj
+        configuration: Release
+        platfom: $(platform)
+        msbuildArguments: /target:BuildCppWithAsan /property:DeployToMonitoringHome=True
+      displayName: Build Native profiler and tests
+
+    - powershell: |        
+        dotnet restore /p:Configuration="Release" /p:Platform="$(platform)" profiler/src/Demos/Datadog.Demos.sln
+      displayName: restore sample applications
+
+    - powershell: |        
+        dotnet build --no-restore -c Release -p:Platform=$(platform) profiler/src/Demos/Datadog.Demos.sln
+      displayName: build sample applications
+
+    - powershell: |
+        # Prepare environment
+        $env:DD_INSIDE_CI=$true
+        $env:DD_HOSTNAME="dd-profiler-dotnet"
+        $env:DD_SERVICE="dd-profiler-dotnet"
+        $env:DD_PROFILING_ENABLED="1"
+
+        # Set profiler deployment and test output dir folder path for the test
+        $test_output_dir="$(System.DefaultWorkingDirectory)\test_output"
+
+        $env:DD_TESTING_OUPUT_DIR=$test_output_dir
+        $env:DD_TRACE_DEBUG="1"
+        $env:DD_PROFILING_LOG_DIR="$test_output_dir\logs"
+        $env:DD_PROFILING_OUTPUT_DIR="$test_output_dir\pprofs"
+        $env:DD_TRACE_ENABLED="0"
```

Comment:
If `DD_TRACE_ENABLED=0`, do we need to download the tracer working directory? Just asking, because it's _huge_ so if we can avoid it, it's worth doing

Reply:
We could download only the artifacts. Because the profiler needs the native loader which is build when building the tracer 😢